modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/store/FilePageStoreManager.java

```diff
@@ -177,19 +168,19 @@ public long allocatePage(int grpId, int partId, byte flags) throws IgniteInterna
      *
      * @param grpName Group name.
      * @param grpId Group ID.
-     * @param partitions Partition number, must be greater than {@code 0} and less {@link PageIdAllocator#MAX_PARTITION_ID} + 1.
+     * @param partitions Partition number, must be greater than {@code 0} and less {@link PageIdAllocator#MAX_PARTITION_ID}.
      * @throws IgniteInternalCheckedException If failed.
      */
     public void initialize(String grpName, int grpId, int partitions) throws IgniteInternalCheckedException {
-        assert partitions > 0 && partitions < MAX_PARTITION_ID + 1 : partitions;
+        assert partitions > 0 && partitions < MAX_PARTITION_ID : partitions;
 
         initGroupDirLock.lock(grpId);
 
         try {
             if (!groupPageStoreHolders.containsKey(grpId)) {
```

Comment:
can we use `groupPageStoreHolders.computeIfAbsent` instead?

Reply:
Creating files can take some more time than just putting an object into memory, from the documentation for **ConcurrentHashMap#computeIfAbsent**, it is recommended to make the function simple and fast so as not to block other threads.